src/main/java/htsjdk/tribble/readers/AsciiLineReader.java

```diff
@@ -36,18 +35,36 @@
  * @author jrobinso
  */
 public class AsciiLineReader implements LineReader, LocationAware {
+    private static final Log log = Log.getInstance(AsciiLineReader.class);
+
     private static final int BUFFER_OVERFLOW_INCREASE_FACTOR = 2;
     private static final byte LINEFEED = (byte) ('\n' & 0xff);
     private static final byte CARRIAGE_RETURN = (byte) ('\r' & 0xff);
 
-    private final PositionalBufferedStream is;
+    private PositionalBufferedStream is;
     private char[] lineBuffer;
     private int lineTerminatorLength = -1;
 
+    protected AsciiLineReader() {};
+
+    /**
+     * TODO: This class implements LocationAware, which requires preservation of virtual file pointers on BGZF inputs.
+     * However, if the inputStream wrapped by this class is a BlockCompressedInputStream, it violates that contract by
+     * wrapping the stream and returning positional file offsets instead.
+     *
+     * @deprecated 8/8/2017 see {@link #from}
```

Comment:
@cmnbroad Yeah, it's really not intended for the users except for them to say "oh, yeah, this is ancient and I shouldn't use it."  A version number wouldn't be a bad idea, but then we have to be sure we guess what version the commit is going into, which is awkward.  Checking the blame is the ultimate guide, but it's a pain to do in bulk.